src/world.hpp

```diff
@@ -126,4 +148,47 @@ class World {
      * @return entity::Mesh the mesh generated
      */
     std::vector<entity::Mesh> get_mesh_greedy() const;
+
+    /**
+     * @brief update all chunk mesh
+     *
+     */
+    void update_all_chunks_mesh();
+
+    // Could mark this inline
+    void update_single_mesh(uint16_t chunk_pos);
+
+    void update_single_mesh(TerrainDim3 tile_sop);
+
+    // set a region to given material, and color
+    void set_tile(uint16_t pos, const terrain::Material* mat, uint8_t color_id);
+
+    void stamp_tile_region(
+        int x_start, int y_start, int z_start, int x_end, int y_end, int z_end,
+        const terrain::Material* mat, std::set<std::pair<int, int>> elements_can_stamp,
+        uint8_t color_id
+    );
+
+    void stamp_tile_region(
+        int x_start, int y_start, int z_start, int x_end, int y_end, int z_end,
+        const terrain::Material* mat, uint8_t color_id
+    );
+
+    inline void
+    qb_save_debug(const std::string& path) {
```

Comment:
```suggestion
    qb_save_debug(const std::string& path) const {
```

Reply:
it's not constant. it changes the data (ya I know its not grate, but i didn't want to copy all of terrain)

Reply:
Why does it change the data, when writing to disk?

Reply:
it sets the color of air aire blocks to a color depending on their chunk and path connected tiles.

Theorietically I could pass a different `get_voxel` function.